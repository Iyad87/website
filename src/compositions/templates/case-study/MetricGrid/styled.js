import styled from 'styled-components';
import breakpoints from 'styles/breakpoints';

export const Grid = styled.ul`
  display: grid;
  position: relative;
  grid-template-columns: 1fr;

  ${breakpoints.tablet`
    grid-template-columns: 1fr 1fr;
  `};
  ${breakpoints.desktop`
    grid-template-columns: repeat(3, 1fr);
  `};
`;
