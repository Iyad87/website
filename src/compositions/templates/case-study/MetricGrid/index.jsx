import React from 'react';
import Metric from 'components/Metric';
import PropTypes from 'prop-types';

import { Grid } from './styled';

const MetricGrid = ({ content }) => (
  <Grid>
    {content &&
      content.map((metric, i) => (
        <Metric key={i} {...metric} max={content.length} />
      ))}
  </Grid>
);

MetricGrid.propTypes = {
  content: PropTypes.arrayOf(PropTypes.shape(Metric.propTypes)).isRequired,
};

export default MetricGrid;
