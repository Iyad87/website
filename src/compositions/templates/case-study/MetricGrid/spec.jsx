import React from 'react';
import { shallow } from 'enzyme';
import MetricGrid from './index';

test('components/Logo snapshot', () => {
  const wrapper = shallow(
    <MetricGrid parameter="reviews" value="2.8" change="20%" />,
  );

  expect(wrapper).toMatchSnapshot();
});
