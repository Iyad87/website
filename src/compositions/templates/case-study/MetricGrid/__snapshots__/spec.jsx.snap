// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Logo snapshot 1`] = `
<styled__Grid>
  <Metric
    change="20%"
    key="reviews"
    parameter="reviews"
    value="2.8"
  />
  <Metric
    change="20%"
    key="reviews"
    parameter="reviews"
    value="2.8"
  />
</styled__Grid>
`;
